sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -121,7 +121,7 @@ class DataFrameSuite extends QueryTest {
     )
   }
 
-  test("self join with aliases") {
+  ignore("self join with aliases") {
```

Comment:
The more I think about this, the more I am worried that we can't make a change this large.  There is no way to express self join queries if we don't handle `.` in column names.  We are also going to break lots of existing user code...